### PR TITLE
add serde helpers crates 

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
related : https://github.com/alloy-rs/alloy/issues/123

### CODE REVIEW

File: crates/serders/src/lib.rs
Comment (mattsse) [inline crates/serders/src/lib.rs:1]: this needs the same header as other crates

**Review by mattsse — CHANGES_REQUESTED** [review]
good start, 
we now want this in all the other rpc crates so we can get rid of the serde_helpers.mod file

**Comment by DoTheBestToGetTheBest** [comment]
> we now want this in all the other rpc crates so we can get rid of the serde_helpers.mod file

do you agree if this can be in separate pr ?

**Comment by mattsse** [comment]
hey @DoTheBestToGetTheBest sorry for the late response

we want this as separate crate yes and then replace the current serde_helpers file in the rpc types crate

currently this duplicates the stuff, let's do it in one go

**Comment by DoTheBestToGetTheBest** [comment]
> hey @DoTheBestToGetTheBest sorry for the late response


no problem (:
> we want this as separate crate yes and then replace the current serde_helpers file in the rpc types crate
> 
> currently this duplicates the stuff, let's do it in one go

okey will finish this

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse does it look great to you ? i don't know how can i run test here

**Comment by DoTheBestToGetTheBest** [comment]
hey boss @mattsse does this looks great to you?

**Comment by DoTheBestToGetTheBest** [comment]
@mattsse hey can you take look here pls

File: crates/rpc-types/Cargo.toml [hunk 1]
```diff
@@ -32,13 +32,27 @@ proptest-derive = { version = "0.4", optional = true }
 # jsonrpsee
 jsonrpsee-types = { version = "0.20", optional = true }
 
+#serde_helper
+alloy-serde.workspace = true
```
Comment (mattsse) [inline crates/rpc-types/Cargo.toml:36]: this should be moved to other alloy deps

File: crates/serders/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+//! Serde helpers for primitive types.
```
Comment (mattsse) [inline crates/serders/src/lib.rs:1]: this needs the header, as used in other crates

File: crates/rpc-types/src/lib.rs [hunk 3]
```diff
@@ -20,7 +20,5 @@
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
 mod eth;
-pub mod serde_helpers;
```
Comment (mattsse) [inline crates/rpc-types/src/lib.rs:23]: we want to keep the reexport for convenience, for now

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc-types/src/lib.rs (hunk 3 above)
Comment (DoTheBestToGetTheBest) [reply crates/rpc-types/src/lib.rs:23]: > we want to keep the reexport for convenience, for now

sorry but i don't understand really what you means here

Comment (mattsse) [reply crates/rpc-types/src/lib.rs:23]: `pub use alloy_serde as serde_helpers`

this way we don't break anything

**Pull request closed without merging**